.github/workflows/add-new-impl-versions.yml

```diff
@@ -0,0 +1,64 @@
+name: Add new implementation versions
+
+on:
+  push:
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: write
+
+jobs:
+  go:
+    runs-on: ubuntu-latest
+    env:
+      DIR: perf/impl/go-libp2p
+      REPO: libp2p/go-libp2p
+      BRANCH: perf/go-libp2p
+    steps:
+      - name: Checkout test-plans
+        uses: actions/checkout@v3
+      - name: Configure git
+        run: |
+          git fetch $BRANCH && git checkout $BRANCH && git rebase $GITHUB_REF -X theirs || git checkout -b $BRANCH
+          git config --global user.email $GITHUB_ACTOR@users.noreply.github.com
+          git config --global user.name $GITHUB_ACTOR
+      - id: go
+        uses: actions/setup-go@v3
+        with:
+          go-version: stable
+      - name: Get the latest version (local)
+        id: local
+        run: ls -d v* | sort -V | tail -n-1 | xargs -I{} echo "version={}" | tee -a $GITHUB_OUTPUT
+        working-directory: ${{ env.DIR }}
+      - name: Get the latest version (remote)
+        id: remote
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: gh api repos/$REPO/releases/latest --jq '.tag_name' | xargs -I{} echo "version={}" | tee -a $GITHUB_OUTPUT
```

Comment:
Have you considered using `go list -m -versions github.com/libp2p/go-libp2p@latest`?